xarray/backends/pnc_.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+
+import numpy as np
+
+from .. import Variable
+from ..core.utils import (FrozenOrderedDict, Frozen)
+from ..core import indexing
+
+from .common import AbstractDataStore, DataStorePickleMixin, BackendArray
+
+
+class PncArrayWrapper(BackendArray):
+
+    def __init__(self, variable_name, datastore):
+        self.datastore = datastore
+        self.variable_name = variable_name
+        array = self.get_array()
+        self.shape = array.shape
+        self.dtype = np.dtype(array.dtype)
+
+    def get_array(self):
+        self.datastore.assert_open()
+        return self.datastore.ds.variables[self.variable_name]
+
+    def __getitem__(self, key):
+        key = indexing.unwrap_explicit_indexer(
+            key, target=self, allow=indexing.BasicIndexer)
```

Comment:
By #1899, we improved the indexing logic for backend arrays, 
to support flexible indexing for all the backend.

Can you update here as similar to the following?
```python
key, np_inds = indexing.decompose_indexer(
    key, self.shape, indexing.IndexingSupport.OUTER_1VECTOR)

array = self.get_array()[key]  # index backend array
if len(np_inds.tuple) > 0:
    array = indexing.NumpyIndexingAdapter(array)[np_inds]  # index the loaded np.ndarray
return array
```

For example, see
https://github.com/pydata/xarray/blob/54468e1924174a03e7ead3be8545f687f084f4dd/xarray/backends/h5netcdf_.py#L18-L32

Here, this
```python
key, np_inds = indexing.decompose_indexer(
    key, self.shape, indexing.IndexingSupport.OUTER_1VECTOR)
```
decomposes the given indexers into those for backend arrays (`key`) and for loaded `np.ndarray` (`np_inds`).

The third argument (`OUTER_1VECTOR`, `OUTER`, `VECTORIZED`, `BASIC`) indicates what kind of indexing the backend support.